pydantic/main.py

```diff
@@ -237,7 +237,10 @@ def model_post_init(self, **kwargs: Any) -> None:
     @typing.no_type_check
     def __setattr__(self, name, value):
         if name in self.__class_vars__:
-            raise ValueError(f'"{self.__class__.__name__}" object has no field "{name}"')
+            raise AttributeError(
```

Comment:
I'll note that producing this error message is the _only_ reason we collect the class_vars in `collect_fields`, and they are only used in the `pydantic._internal._model_construction.complete_model_class` call to `collect_fields` (one of four calls).

That said, I still think it makes sense to collect them there, and to produce this error message.